### PR TITLE
sg: add commands to override logger levels for individual services

### PR DESCRIPTION
**Pull request by coury-clark**
Adds new options to the `run-set` command allowing you to override the logger level for specific services.
`--debug`
`--info`
`--warn`
`--error`
`--crit`

`sg run-set --debug=gitserver --error=enterprise-worker,enterprise-frontend enterprise-codeinsights`

### CODE REVIEW

File: dev/sg/main.go [hunk 1]
```diff
@@ -354,9 +359,56 @@ func runSetExec(ctx context.Context, args []string) error {
...
+	}
 	return run.Commands(ctx, globalConf.Env, cmds...)
 }
 
+// enrichWithLogLevels will add any logger level overrides to a given command if they have been specified.
+func enrichWithLogLevels(cmd *run.Command, overrides map[string]string) {
+	logLevelVariable := "SRC_LOG_LEVEL"
+
+	if level, ok := overrides[cmd.Name]; ok {
+		out.WriteLine(output.Linef("", output.StyleWarning, "Setting log level: %s for command %s.", level, cmd.Name))
```
Comment (mrnugget) [inline dev/sg/main.go:374]: We need to redo the whole color thing anyway (find a fixed set of colors, use them consistently, etc.) but maybe this?

```suggestion
		out.WriteLine(output.Linef("", output.StylePending, "Setting log level: %s for command %s.", level, cmd.Name))
```

File: dev/sg/main.go [hunk 2]
```diff
@@ -313,7 +318,7 @@ func runSetExec(ctx context.Context, args []string) error {
 		return flag.ErrHelp
 	}
 
-	if len(args) != 1 {
+	if len(args) > 2 {
```
Comment (mrnugget) [inline dev/sg/main.go:321]: Out of curiosity: does that mean that the flag counts as an arg? That makes me wonder whether we can't check which flags where given.

**Review by mrnugget — APPROVED** [review]
That's super handy! Thanks!

Can you also add this to the `README.md`? Just as an example under the `run-set` explanations.

File: dev/sg/main.go (hunk 1 above)
Comment (coury-clark) [reply dev/sg/main.go]: Sure thing!

File: dev/sg/main.go (hunk 2 above)
Comment (coury-clark) [reply dev/sg/main.go:321]: It seems so - it also picks up these flags as inline args if you put them at the end.

**Pull request merged**